crates/api/src/site/registration_applications/tests.rs

```diff
@@ -0,0 +1,435 @@
+    .registration_mode(Some(RegistrationMode::RequireApplication))
+    .site_setup(Some(true))
+    .build();
+  let local_site = LocalSite::create(pool, &local_site_form).await.unwrap();
+
+  // Required to have a working local SiteView when updating the site to change email verification
+  // requirement or registration mode
+  let rate_limit_form = LocalSiteRateLimitInsertForm::builder()
+    .local_site_id(local_site.id)
+    .build();
+  LocalSiteRateLimit::create(pool, &rate_limit_form)
+    .await
+    .unwrap();
+
+  Ok((inserted_instance, admin_local_user_view))
+}
+
+async fn signup(
+  pool: &mut DbPool<'_>,
+  instance_id: InstanceId,
+  name: &str,
+  email: Option<&str>,
+) -> LemmyResult<(LocalUser, RegistrationApplication)> {
+  let person_insert_form = PersonInsertForm::test_form(instance_id, name);
+  let person = Person::create(pool, &person_insert_form).await?;
+
+  let local_user_insert_form = match email {
+    Some(email) => LocalUserInsertForm {
+      email: Some(email.to_string()),
+      email_verified: Some(false),
+      ..LocalUserInsertForm::test_form(person.id)
+    },
+    None => LocalUserInsertForm::test_form(person.id),
+  };
+
+  let local_user = LocalUser::create(pool, &local_user_insert_form, vec![]).await?;
+
+  let application_insert_form = RegistrationApplicationInsertForm {
+    local_user_id: local_user.id,
+    answer: "x".to_string(),
+  };
+  let application = RegistrationApplication::create(pool, &application_insert_form).await?;
+
+  Ok((local_user, application))
+}
+
+#[allow(clippy::unwrap_used)]
+async fn get_application_statuses(
+  context: &Data<LemmyContext>,
+  admin: LocalUserView,
+) -> LemmyResult<(
+  Json<GetUnreadRegistrationApplicationCountResponse>,
+  Json<ListRegistrationApplicationsResponse>,
+  Json<ListRegistrationApplicationsResponse>,
+)> {
+  let application_count =
+    get_unread_registration_application_count(context.reset_request_count(), admin.clone()).await?;
+
+  let unread_applications = list_registration_applications(
+    Query::from_query("unread_only=true").unwrap(),
+    context.reset_request_count(),
+    admin.clone(),
+  )
+  .await?;
+
+  let all_applications = list_registration_applications(
+    Query::from_query("unread_only=false").unwrap(),
+    context.reset_request_count(),
+    admin,
+  )
+  .await?;
+
+  Ok((application_count, unread_applications, all_applications))
+}
+
+#[allow(clippy::indexing_slicing)]
+#[allow(clippy::unwrap_used)]
+#[tokio::test]
+#[serial]
+async fn test_application_approval() -> LemmyResult<()> {
+  let context = LemmyContext::init_test_context().await;
+  let pool = &mut context.pool();
+
+  let (instance, admin_local_user_view) = create_test_site(&context).await?;
+
+  // Non-unread counts unfortunately are duplicated due to different types (i64 vs usize)
+  let mut expected_application_count = 0;
+  let mut expected_unread_applications_len = 0;
+  let mut expected_applications_len = 0;
```

Comment:
It should be possible to deduplicate the 2 non-unread counts by using `u8` for the type, and in each `assert_eq` use `.into()`. You might need to make it the 2nd argument instead of the 1st argument to make type inference work.

Reply:
thanks, but even that doesn't make type inference work:

```rs
let mut expected_unread_applications = 0u8;
// [...]
assert_eq!(
  application_count.registration_applications,
  expected_unread_applications.into(),
);
```

```
error[E0283]: type annotations needed
   --> crates/api/src/site/registration_applications/tests.rs:175:34
    |
175 |     expected_unread_applications.into(),
    |                                  ^^^^
    |
    = note: cannot satisfy `_: std::convert::From<u8>`
    = note: required for `u8` to implement `std::convert::Into<_>`
help: try using a fully qualified path to specify the expected types
    |
175 |     <u8 as std::convert::Into<T>>::into(expected_unread_applications),
    |     ++++++++++++++++++++++++++++++++++++                            ~
```

explicit annotation does work:
```rs
assert_eq!(
  application_count.registration_applications,
  <u8 as Into<i64>>::into(expected_total_applications),
);
```

Reply:
Use `i64::from`

Reply:
done